heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -179,6 +179,11 @@ export default class JestPlugin implements IHeftTaskPlugin<IJestPluginOptions> {
   ): void {
     const { parameters } = taskSession;
 
+    // set env variable if not already set
+    if (!process.env.NODE_ENV) {
+      process.env.NODE_ENV = 'test';
+    }
```

Comment:
This should really happen as part of running, and the value should get reverted as soon as the run completes.

Reply:
yes, that makes sense @dmichon-msft. I thought of it already but then I decided to do it like this since this is exactly how it is done in jest, no cleanup no nothing. 

I guess you are referring to the part where the `runCLI` is triggered?

Reply:
Here for non-watch: https://github.com/microsoft/rushstack/pull/4357/files#diff-41aabbe4383e1298817c83a6f0fb66bbb3a446578445ca53e3a5c819af286154L276-L282
Here for watch: https://github.com/microsoft/rushstack/pull/4357/files#diff-41aabbe4383e1298817c83a6f0fb66bbb3a446578445ca53e3a5c819af286154L492-L514

The reason is that unlike when running Jest from the CLI, when running in Heft it shares the process with other tools that might care about the value of process.env.NODE_ENV.

Reply:
please check it again. i also skipped te test since i hae no experience how to test the actual plugin